### PR TITLE
Some misc helpers around the E2EE support

### PR DESCRIPTION
**Pull request by poljar**
Some more commits from #2621.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/store/mod.rs [hunk 1]
```diff
@@ -671,7 +671,7 @@ pub struct BackupKeys {
 
 /// A struct containing private cross signing keys that can be backed up or
 /// uploaded to the secret store.
-#[derive(Zeroize)]
+#[derive(Zeroize, Default)]
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/store/mod.rs:674]: std traits first?

```suggestion
#[derive(Default, Zeroize)]
```

**Review by jplatte — APPROVED** [review]

File: crates/matrix-sdk/tests/integration/client.rs [hunk 2]
```diff
@@ -503,3 +503,73 @@ async fn marking_room_as_dm() {
...
+
+#[cfg(feature = "e2e-encryption")]
+#[async_test]
+async fn cross_signing_status() {
+    let (client, server) = logged_in_client().await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/unstable/keys/device_signing/upload"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({})))
+        .expect(1)
```
Comment (bnjbvr) [inline crates/matrix-sdk/tests/integration/client.rs:538]: drive-by: expect has no effect if not calling `server.verify()` at the end, right?

Comment (bnjbvr) [reply crates/matrix-sdk/tests/integration/client.rs:538]: (disabled auto-merge based on that comment)

Comment (poljar) [reply crates/matrix-sdk/tests/integration/client.rs:538]: Added now, rebased and squashed things as well.
